### PR TITLE
Api provides functionality for authenticated users to create events

### PR DESCRIPTION
**Pull request by ghost**
## [PT Story](https://www.pivotaltracker.com/story/show/172391432)
```
As an api
In order for an authenticated user to join events
I need to allow only authenticated users access to join an event.
```
## Changes proposed in this pull request:
* User model created and configured
* Creates user factory
* Makes sure that a user is logged in before creating an event
## What I have learned working on this feature:
* Refreshed the knowledge of creating a login function
* Learning more about devise token auth

### CODE REVIEW

**Review by dernathul — APPROVED** [review]

**Review by tochman — CHANGES_REQUESTED** [review]
The ability to login or sign up is NOT a feature.

File: db/schema.rb [hunk 1]
```diff
@@ -47,10 +47,8 @@
     t.datetime "confirmed_at"
     t.datetime "confirmation_sent_at"
     t.string "unconfirmed_email"
-    t.string "name"
-    t.string "nickname"
-    t.string "image"
     t.string "email"
+    t.boolean "authenticated"
```
Comment (oliverochman) [inline db/schema.rb]: not needed, remove this attribute

File: spec/requests/user_can_create_an_event_spec.rb [hunk 2]
```diff
@@ -48,4 +51,28 @@
       expect(JSON.parse(response.body)['message']).to eq 'Event was NOT created.'
```
Comment (oliverochman) [inline spec/requests/user_can_create_an_event_spec.rb]: Why was it not created?

File: db/schema.rb (hunk 1 above)
Comment (hmvitous) [reply db/schema.rb]: We removed it

File: spec/requests/user_can_create_an_event_spec.rb (hunk 2 above)
Comment (hmvitous) [reply spec/requests/user_can_create_an_event_spec.rb]: We have edited that message to be more specific

File: spec/requests/user_can_login_spec.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    it 'returns error message' do
+      expect(response_json['errors']).to eq ['Invalid login credentials. Please try again.']
+    end
+  end
+
+  describe 'with invalid email' do
+    before do
+      post '/api/auth/sign_in',
+           params: {
+             email: 'wrong@email.com',
```
Comment (dernathul) [inline spec/requests/user_can_login_spec.rb:58]: wrong email seems pretty right email to me...you should provide another without the @ symbol

**Comment by ghost** [comment]
lgmt

**Review by dernathul — APPROVED** [review]

**Pull request merged**